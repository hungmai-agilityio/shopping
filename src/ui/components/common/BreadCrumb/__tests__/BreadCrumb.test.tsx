import { render, screen } from '@testing-library/react';

// Mocks
import { mockBreadCrumb } from '@/mocks';

// Components
import Breadcrumb from '@/ui/components/common/BreadCrumb';

describe('BreadCrumb component', () => {
  const breadcrumbItems = [
    { label: 'Home', href: '/' },
    { label: 'Shopping', href: '/shopping' },
    { label: 'Electronics', href: '/shopping/electronics' },
    { label: 'Phones', href: '' }
  ];

  it('render with default props', () => {
    render(<Breadcrumb items={mockBreadCrumb} />);

    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Products')).toBeInTheDocument();
    expect(screen.getByText('Phones')).toBeInTheDocument();
  });

  it('should show truncated version when there are more than 3 items', () => {
    render(<Breadcrumb items={breadcrumbItems} separator=">" />);

    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('...')).toBeInTheDocument();
    expect(screen.getByText('Electronics')).toBeInTheDocument();
    expect(screen.getByText('Phones')).toBeInTheDocument();
  });

  it('should be render match to snapshot', () => {
    const { container } = render(<Breadcrumb items={mockBreadCrumb} />);

    expect(container).toMatchSnapshot();
  });
});
