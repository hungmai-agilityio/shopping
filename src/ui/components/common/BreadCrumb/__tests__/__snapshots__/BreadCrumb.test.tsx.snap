// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb component should be render match to snapshot 1`] = `
<div>
  <nav
    aria-label="breadcrumb"
    class="flex items-center space-x-2 text-black"
  >
    <span
      class="flex items-center"
    >
      <a
        class="hover:underline"
        href="/"
      >
        Home
      </a>
      <span
        class="mx-2 text-black"
      >
        /
      </span>
    </span>
    <span
      class="flex items-center"
    >
      <a
        class="hover:underline"
        href="/products"
      >
        Products
      </a>
      <span
        class="mx-2 text-black"
      >
        /
      </span>
    </span>
    <span
      class="flex items-center"
    >
      <span
        class="font-bold text-black"
      >
        Phones
      </span>
    </span>
  </nav>
</div>
`;
